### ISSUE TITLE
Contact for a few question

### ISSUE THREAD

**Issue by warthurton** [denolfe/AutoHotkey#6]
I have a few questions about the scripts as I have been using and customizing for a while.  Is there a way that I can contact you outside of the issue list?

**Comment by denolfe** [comment]
I'll simply label this issue as a question, and I can answer them here.

EDIT: Unless there is some particular reason why you wouldn't want to do it here.

**Comment by warthurton** [comment]
I didn't want to spam the issues for a non-issue, but this will work fine.

- Work.ahk is designed to run with administrator privileges and I'm curious how have you gotten around some programs (Outlook, VS code, etc.) not preferring this?  I updated the Show.ahk to use a utility called hstart, and it can force non-elevated run.  This app works well but adds a tiny delay into launching due to the way it handles it.
- Have you ever figured out a way to detect which keyboard is active?  Or even map them out.  Trial and error worked for me but it it doesn't handle my laptop well.  I had some ideas but was hoping you found some proven way.  I could detect the presence of a USB keyboard using some yet to be determined method (monitor the event log using windows task scheduler is one I've read about).  Hotkeys could be enabled/disabled depending on the keyboard.
- Do you still use MyMenu.ahk?  I had to update the #Include, so I was curious if you had abandoned it for some reason
- What did you use to make the lovely keyboard map image?  I'd like to update for my bindings and have little graphics skills.

Hopefully not too overwhelming.  Thanks for the very helpful starting point.

	if elevated
	{
		Hstart := "C:\Utils\hstart\hstart.exe /SHELL /NOUAC """"" . exe . """ " . params . """"
	}
	Else
	{
		Hstart := "C:\Utils\hstart\hstart.exe /SHELL /NONELEVATED """"" . exe . """ " . params . """"
	}

Then later

    Run, %Hstart%,,UserErrorLevel

**Comment by warthurton** [comment]
if elevated
	{
		Hstart := "C:\Utils\hstart\hstart.exe /SHELL /NOUAC """"" . exe . """ " . params . """"
	}
	Else
	{
		Hstart := "C:\Utils\hstart\hstart.exe /SHELL /NONELEVATED """"" . exe . """ " . params . """"
	}

Then later

    Run, %Hstart%,,UserErrorLevel

**Comment by denolfe** [comment]
- The actually have not tried to find a solution for applications that don't like to run as admin. I tend to just run them as admin anyway currently. Very interesting solution you have going. I may take a look.
- I haven't had a need for treating input devices differently. I like everything to be enabled on every keyboard.
- You are correct. I no longer utilize `MyMenu.ahk`. I was hardly ever utilizing the utility applications in the menu.
- The graphic was made using [Keyboard Layout Editor](http://www.keyboard-layout-editor.com/). Here is the [link](https://gist.github.com/denolfe/46a335c24d56b3065040302dedfb8a86) to my layout.

**Comment by warthurton** [comment]
I've used hstart for years as I can't permanently disable UAC due to some group policies and it allows for elevating w/o UAC prompt, but recently have been using to for non-administrator run.  https://www.ntwind.com/software/hstart.html

I've also experimented with http://www.nirsoft.net/utils/advanced_run.html and there seems to be a way without using a tool, just windows batch https://superuser.com/a/981202/8790

I'll clean up my private repo and make it a true fork so that I can show you some of the things I have worked on.  

Thank you for the link to Keyboard editor.
